tests/test_alert_actions.py

```diff
@@ -0,0 +1,242 @@
+            "target": "N/A",
+            "severity": "LOW",
+            "ruleId": "cab2d5ee-a512-45b1-8848-809327033048",
+            "ruleSource": "Alerting",
+            "id": "11111111-9724-4005-b848-76af488cf5e2",
+            "createdAt": "2021-05-13T16:51:35.4259080Z",
+            "state": "OPEN",
+        },
+        {
+            "type$": "ALERT_SUMMARY",
+            "tenantId": "11111111-af5b-4231-9d8e-df6434da4663",
+            "type": "FED_COMPOSITE",
+            "name": "File Upload Alert",
+            "description": "Alert on any file upload events",
+            "actor": "barney.frankenberry@chocula.com",
+            "actorId": "987210998131391466",
+            "target": "N/A",
+            "severity": "MEDIUM",
+            "ruleId": "962a6a1c-54f6-4477-90bd-a08cc74cbf71",
+            "ruleSource": "Alerting",
+            "id": "1111111-555f-4880-8909-f5679448e67c",
+            "createdAt": "2021-05-13T16:51:35.3465540Z",
+            "state": "OPEN",
+        },
+    ],
+    "totalCount": 2,
+    "problems": [],
+}
+
+
+class TestCode42AlertsConnector(object):
+    def test_handle_action_when_alert_detail_calls_get_with_expected_args(
+        self, mock_py42_client
+    ):
+        param = {"alert_id": "123-456-7890"}
+        connector = _create_get_alert_details_connector(mock_py42_client)
+
+        connector.handle_action(param)
+
+        mock_py42_client.alerts.get_details.assert_called_once_with([param["alert_id"]])
+
+    def test_handle_action_when_alert_detail_adds_response_to_data(
+        self, mock_py42_with_alert_details
+    ):
+        param = {"alert_id": "123-456-7890"}
+        connector = _create_get_alert_details_connector(mock_py42_with_alert_details)
+        connector.handle_action(param)
+        assert_successful_single_data(
+            connector, _MOCK_ALERT_DETAIL_RESPONSE["alerts"][0]
+        )
+
+    def test_handle_action_when_alert_detail_adds_info_to_summary(
+        self, mock_py42_with_alert_details
+    ):
+        param = {"alert_id": "123-456-7890"}
+        connector = _create_get_alert_details_connector(mock_py42_with_alert_details)
+        connector.handle_action(param)
+        expected_summary = {
+            "username": _MOCK_ALERT_DETAIL_RESPONSE["alerts"][0]["actor"],
+            "user_id": _MOCK_ALERT_DETAIL_RESPONSE["alerts"][0]["actorId"],
+        }
+        assert_successful_summary(connector, expected_summary)
+
+    def test_handle_action_when_search_alerts_calls_search(self, mock_py42_client):
+        param = {
+            "username": "gilbert.lequiche@finance.biz",
+            "start_date": "2031-01-01",
+            "end_date": "2031-06-01",
+            "alert_state": "OPEN",
+        }
+        connector = _create_search_alerts_connector(mock_py42_client)
+        connector.handle_action(param)
+        mock_py42_client.alerts.search.assert_called_once()
+
+    def test_handle_action_when_search_alerts_and_all_params_missing_sets_error_message(
```

Comment:
These tests are all titled that they are testing `sets_error_message`, but they all just check the _status_ is a an error. Might be a good time to add an `assert_failure_message` helper?